src/nodes.coffee

```diff
@@ -1121,10 +1120,17 @@ exports.Obj = class Obj extends Base
     @lhs = yes
 
     for prop in @properties
+      # Check for reserved words.
+      message = isUnassignable prop.unwrapAll().value
```

Comment:
This took me perhaps the most time to figure out. It _is_ being checked in `Assign`, but `Assign` never gets invoked for function parameter destructuring like `({a}) ->`.